compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -150,6 +150,59 @@ impl AttemptLocalParseRecovery {
     }
 }
 
+/// Information for emitting suggestions and recovering from
+/// C-style `i++`, `--i`, etc.
+#[derive(Debug, Copy, Clone)]
+struct IncDecRecovery {
+    /// Is this increment/decrement its own statement?
+    ///
+    /// This is `None` when we are unsure.
+    standalone: Option<bool>,
```

Comment:
nit: it might be more self-documenting if this was an enum with three variants, I skipped over the comment at first and it took a second to understand what situation `None` corresponded to

Reply:
Done, good idea.